animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,117 @@
+from libc.string cimport memset
+from cpython.mem cimport PyMem_Malloc, PyMem_Realloc, PyMem_Free
+from ... data_structures cimport (
+    Mesh, Vector3DList, EdgeIndicesList, PolySpline,
+    VirtualDoubleList, FloatList
+)
+
+def splinesFromBranches(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii):
+    cdef int i, j
+    cdef int edgesAmount = len(edges)
+    cdef int verticesAmount = len(vertices)
+
+    # Compute how many neighbour each vertex have.
+    cdef int *neighboursAmounts = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
+    memset(neighboursAmounts, 0, verticesAmount * sizeof(int))
+    for i in range(edgesAmount):
+        neighboursAmounts[edges.data[i].v1] += 1
```

Comment:
this could raise an segmentation fault if there aren't enough vertices.
The node itself should check wether the inpu is valid (max vertex index of an edge is smaller than the number of vertices, like in the original code)

Reply:
Should the node itself have error handling then?

Reply:
yes, I think the node should do the check. "cutils" usually should not have to make expensive checks, they can expect the data to valid.
Nodes on the other hand can not expect that.

Reply:
Alright.